src/csharp/Microsoft.Spark/Sql/Types/DataType.cs

```diff
@@ -56,10 +55,10 @@ public string Json
         {
             get
             {
-                object jObject = (JsonValue is JObject) ?
-                    ((JObject)JsonValue).SortProperties() :
+                object jObject = (JsonValue is JsonElement jsonElement) ?
+                    jsonElement.SortProperties() :
                     JsonValue;
-                return JsonConvert.SerializeObject(jObject, Formatting.None);
+                return JsonSerializer.Serialize(jObject);
```

Comment:
Making `JsonElement` go through a sort phase and then a serialize phase seems inefficient since you are doing the work twice.

When will `JsonValue` not be a `JsonElement`? Maybe only call the serializer for that case and otherwise, just have a SortProperties helper that returns the string itself.